google/cloud/aiplatform/prediction/local_endpoint.py

```diff
@@ -0,0 +1,469 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import logging
+from pathlib import Path
+import requests
+import time
+from typing import Any, Dict, List, Optional, Sequence
+
+from google.auth.exceptions import GoogleAuthError
+
+from google.cloud.aiplatform import initializer
+from google.cloud.aiplatform.constants import prediction
+from google.cloud.aiplatform.docker_utils import run
+from google.cloud.aiplatform.docker_utils.errors import DockerError
+from google.cloud.aiplatform.utils import prediction_utils
+
+_logger = logging.getLogger(__name__)
+
+_DEFAULT_CONTAINER_READY_TIMEOUT = 300
+_DEFAULT_CONTAINER_READY_CHECK_INTERVAL = 1
+
+_GCLOUD_PROJECT_ENV = "GOOGLE_CLOUD_PROJECT"
+
+
+class LocalEndpoint:
```

Comment:
Same logic as the comment on LocalModel: If LocalEndpoint is supposed to be analogous to the Endpoint SDK type, then it should have the same interface. This should be a subclass of Endpoint and override methods its methods. Otherwise, a divergence between Endpoint and LocalEndpoint is all but guaranteed.

Reply:
LocalEndpoint is also "the local version" of the Endpoint but they are not able to be bridged directly (https://github.com/googleapis/python-aiplatform/blob/custom-prediction-routine/google/cloud/aiplatform/prediction/docs/cpr_cuj.svg). See my comments for LocalModel above.

Reply:
Sorry, but this is unfortunately not optional. We can't have two different interfaces for Endpoint.